tests/extended_temporal_memory/extensive_etm_test.py

```diff
@@ -0,0 +1,1340 @@
+
+    self._testTM(testProximalInput,
+                 activeExternalCellsSequence=externalInputA[1:],
+                 activeApicalCellsSequence=feedbackB)
+    # no burst at last step
+    unpredictedActiveColumns = self.tm.mmGetTraceUnpredictedActiveColumns().data
+    self.assertGreaterEqual(len(unpredictedActiveColumns[-1]), min(self.w))
+    # many extra predictions
+    predictedInactiveColumns = self.tm.mmGetTracePredictedInactiveColumns().data
+    self.assertGreaterEqual(len(predictedInactiveColumns[-1]), min(self.w))
+
+
+  def testO3(self):
+    """Joint external and apical inputs, with incorrect external prediction.
+
+    Train on ABCDE / PQRST with F1, XBCDF / PQRST with F2.
+    Test with BCDE / QRXT. With feedback F1, burst at last step, as prediction
+    from external input is incorrect.
+    """
+    self.init()
+
+    numbers = self.sequenceMachine.generateNumbers(1, 5)
+    proximalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+
+    testProximalInput = proximalInputA[1:]
+
+    numbers[0] = max(numbers) + 10
+    numbers[-2] = max(numbers) + 10
+    proximalInputB = self.sequenceMachine.generateFromNumbers(numbers)
+
+    numbers = self.sequenceMachine.generateNumbers(1, 5)
+    externalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+    numbers[-3] = max(numbers) + 10
+    testExternalInput = self.sequenceMachine.generateFromNumbers(numbers)[1:]
+
+    feedbackA = self.generateFeedbackSequence(len(numbers))
+    feedbackB = self.generateFeedbackSequence(len(numbers))
+
+    # start training with random feedback
+    self.feedTM(proximalInputA,
+                activeExternalCellsSequence=externalInputA,
+                activeApicalCellsSequence=self.generateRandomFeedback(len(numbers)))
+    self.feedTM(proximalInputB,
+                activeExternalCellsSequence=externalInputA,
+                activeApicalCellsSequence=self.generateRandomFeedback(len(numbers)))
+
+    for _ in xrange(30):
+      self.feedTM(proximalInputA,
+                  activeExternalCellsSequence=externalInputA,
+                  activeApicalCellsSequence=feedbackA,
+                  formInternalConnections=True)
+      self.feedTM(proximalInputB,
+                  activeExternalCellsSequence=externalInputA,
+                  activeApicalCellsSequence=feedbackB,
+                  formInternalConnections=True)
+
+    self._testTM(testProximalInput,
+                 activeExternalCellsSequence=testExternalInput,
+                 activeApicalCellsSequence=feedbackA)
+    # burst at last step
+    unpredictedActiveColumns = self.tm.mmGetTraceUnpredictedActiveColumns().data
+    self.assertGreaterEqual(len(unpredictedActiveColumns[-1]), min(self.w))
+
+
+  def testO4(self):
+    """Joint external / apical inputs, with correct feedback.
+
+    Train on ABCDE / PQRST with F1, XBCDEF / PQRST with F2.
+    Test with BCDE / QRST. With feedback F1, no burst nor extra (feedback
```

Comment:
This is a good test, but do you want to have a unique ending for the first sequence?